subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -109,124 +109,136 @@ static int update_job_execution(struct mqtt_client *const client,
 	return ret;
 }
 
-
-static int aws_fota_on_publish_evt(struct mqtt_client *const client,
-				   const u8_t *topic,
-				   u32_t topic_len,
-				   u32_t payload_len)
+static int get_job(struct mqtt_client *const client, u32_t payload_len)
 {
-	int err;
+	int err = get_published_payload(client, payload_buf, payload_len);
 
-	LOG_INF("Received topic: %s", log_strdup(topic));
+	if (err) {
+		LOG_ERR("error when getting the payload: %d", err);
+		return err;
+	}
 
-	bool is_get_next_topic = aws_jobs_cmp(get_topic, topic, topic_len, "");
-	bool is_notify_next_topic = aws_jobs_cmp(notify_next_topic, topic,
-						 topic_len, "");
+	/* Check if message received is a job. */
+	err = aws_fota_parse_notify_next_document(payload_buf,
+			payload_len, job_id,
+			hostname, file_path,
+			&doc_version_number);
+	if (err < 0) {
+		LOG_ERR("Error when parsing the json: %d", err);
+		return err;
+	} else  if (err == 1) {
+		/* Empty job execution object */
+		LOG_INF("No job execution object found: %s",
+				log_strdup(payload_buf));
+		return 1;
+	}
 
-	if (is_notify_next_topic || is_get_next_topic) {
-		err = get_published_payload(client, payload_buf, payload_len);
-		if (err) {
-			LOG_ERR("Error when getting the payload: %d", err);
-			return err;
-		}
-		/* Check if message received is a job. */
-		err = aws_fota_parse_notify_next_document(payload_buf,
-							  payload_len, job_id,
-							  hostname, file_path);
+	LOG_DBG("Job ID: %s", job_id);
+	LOG_DBG("hostname: %s", hostname);
+	LOG_DBG("file_path %s", file_path);
+	LOG_DBG("doc_version_number: %d ", doc_version_number);
 
-		if (err < 0) {
-			LOG_ERR("Error when parsing the json: %d", err);
-			return err;
-		} else  if (err == 1) {
-			LOG_INF("Got only one field: %s",
-				log_strdup(payload_buf));
-			return 1;
-		}
+	/* Subscribe to update topic to recive feedback on whether an update is
+	 * accepted or not.
+	 */
+
+	err = aws_jobs_subscribe_topic_update(client, job_id,  update_topic);
+	if (err) {
+		LOG_ERR("Error when subscribing job_id_update: %d", err);
+		return err;
+	}
+
+	/* Set fota_state to DOWNLOAD_FIRMWARE, when we are subscribed
+	 * to job_id topics we will try to publish and if accepted we
+	 * can start the download
+	 */
+
+	fota_state = DOWNLOAD_FIRMWARE;
+
+	return 1;
+}
 
-		/* Unsubscribe from notify_next_topic to not recive more jobs
-		 * while processing the current job.
+static int update_accepted(struct mqtt_client *const client, u32_t payload_len)
```

Comment:
Short doc string for `update_accepted` as well.

Reply:
done